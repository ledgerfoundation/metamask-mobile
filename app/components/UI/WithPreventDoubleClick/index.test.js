import React from 'react';
import Button from '../Button';
import { withPreventDoubleClickErrorMsg, withPreventDoubleClick } from './';

const noop = () => ({});

describe('withPreventDoubleClick', () => {
	it('should render correctly', () => {
		const ButtonWithOnPress = () => <Button onPress={noop} />;
		const WithPreventDoubleClickButton = withPreventDoubleClick(ButtonWithOnPress);
		const wrapper = <WithPreventDoubleClickButton onPress={noop} />;
		expect(wrapper).toMatchSnapshot();
	});

	it('should be wrapped and named correctly', () => {
		const ButtonWithOnPress = () => <Button onPress={noop} />;
		const name = `withPreventDoubleClick(${ButtonWithOnPress.name})`;
		const WithPreventDoubleClickButton = withPreventDoubleClick(ButtonWithOnPress);
		const wrapper = <WithPreventDoubleClickButton onPress={noop} />;
		expect(wrapper.type.displayName).toBe(name);
	});

	it('should throw an error if Component is not provided', () => {
		try {
			withPreventDoubleClick();
		} catch (e) {
			expect(e.message).toBe(withPreventDoubleClickErrorMsg);
		}
	});
});
