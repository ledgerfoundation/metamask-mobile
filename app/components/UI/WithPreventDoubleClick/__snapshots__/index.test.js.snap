// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPreventDoubleClick should render correctly 1`] = `
<withPreventDoubleClick(ButtonWithOnPress)
  onPress={[Function]}
/>
`;
